core/src/com/unciv/models/ruleset/Building.kt

```diff
@@ -615,7 +616,7 @@ class Building : RulesetStatsObject(), INonPerpetualConstruction {
 
         for ((resource, amount) in getResourceRequirements())
             if (civInfo.getCivResourcesByName()[resource]!! < amount) {
-                rejectionReasons.add(RejectionReason.ConsumesResources.toInstance(resource.getConsumesAmountString(amount)))
+                rejectionReasons.add(RejectionReason.ConsumesResources.toInstance(resource.getNeedMoreAmountString(amount)))
```

Comment:
I'd make this `getNeedMoreAmountString(amount - civInfo.getCivResourcesByName()[resource]!!`, so you get the amount of resources you still need extra